### ISSUE TITLE
[QUESTION] PakjeGemak location name in transactional emails

### ISSUE THREAD

**Issue by watista** [tig-nl/postnl-magento2#221]
### TL;DR ###
Is it possible to add the name of the PakjeGemak location to the transactional emails send to customers?

### Situation ###
I am using the main layout of the default magento transactional emails. With the order confirmation email the 'Billing Address' and 'Shipping Address' is displayed along with some other information about the order.

When a customer chooses for PakjeGemak delivery the address of the 'Shipping Address' table in the transactional email is changed to the address of the PakjeGemak location. The name of the customer is still the same.

### Image example ###
In the image below the 'Billing Address' field is named 'Factuuradres' and the 'Shipping Address' is named 'Verzendinformatie':
![order-email](https://user-images.githubusercontent.com/37804556/101358522-42660280-389b-11eb-9275-466a7a8ce26f.PNG)

### Question ###
Is it possible to display the name of the PakjeGemak location under the customer name in the 'Shipping Address' table in the transactional emails? Or is there another possible way to indicate in the email that the order is send with PakjeGemak?

### Extra info ###
Magento 2.3.5-p1
PostNL 1.9.5
2-Step Checkout Luma

I've tried multiple variables that i found in the docs (https://docs.tig.nl/pages/viewpage.action?pageId=8553643) but none of them seem to give the PakjeGemak location name.

Thanks in advance!

**Comment by tig-dennisvanderhammen** [comment]
Hi @watista ,

Interesting question! I'm really wondering about the possibilities here as well.
First of all. I noticed that the getTemplateVars function in \TIG\PostNL\Helper\Tracking\Mail is private, making it hard for people to write a plugin on it. I'll create a pull request internally to change this to a public function, allowing people to create an afterPlugin, to add their own template variables.

As for the order email, I'll have to look into it another time, but as a solution for now, maybe you could create a function on this event: email_order_set_template_vars_before, which happens in  magento/module-sales/Model/Order/Email/Sender/OrderSender.php.

The name of the address gets stored in the Company field of the Address. You could either change the TransportObject, depending on if it's a Post Office location, or maybe there are options to have conditional checks on the templates (My knowledge about email templates are a bit dated).

For now I'll put your issue on the backlog, as this seems like a cool feature in my opinion, and the current way might cause some confusion with clients. Please note that PostNL decides what features get implemented.

**Comment by watista** [comment]
Hi @tig-dennisvanderhammen,

Thanks for your quick reply.

I will try your solution and post the results here. I must admit that my experience with custom Magento modules and functions is not that much so it could take a couple of days.

Please keep me informed about the backlog feature and the decision of PostNL.

Regards,

Wouter

**Comment by fourtwenty** [comment]
Is there any update about this?

**Comment by tig-jeffreybranderhorst** [comment]
Hi @fourtwenty,

The default template of the order confirmation email should display the address and name of the pickup point in the shipping address.  Do you want to display is in the order confirmation email or in another email?

Have a great day,
Jeffrey

**Comment by fourtwenty** [comment]
never mind, just tested it and it displays the name of the Pickup point.

**Comment by tig-jeffreybranderhorst** [comment]
Hi @fourtwenty,

Indeed! Thank you for the confirmation! 

Have a great day,
Jeffrey

**Comment by tig-jeffreybranderhorst** [comment]
Hi @watista,

In the newest version of the extension, the address in the confirmation email should be correct and show the pickup point. Because it is fixed now in the newest version, I will close this issue.  If you still got question, don't hesitate to contact us on support@tig.nl

Have a great day,
Jeffrey

**Comment by watista** [comment]
Hi @tig-jeffreybranderhorst,

Thanks for the update and for adding this feature!
